src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js

```diff
@@ -107,15 +113,54 @@ describe('AzureComputeModal', () => {
       saturnWorkspaceName: defaultModalProps.workspace.workspace.name
     }
     expect(runtimeFunc).toHaveBeenCalledWith(defaultModalProps.workspace.workspace.workspaceId, expect.anything())
-    expect(createFunc).toHaveBeenCalledWith(expect.objectContaining({
+    expect(createFunc).toHaveBeenCalledWith({
       labels,
       disk: expect.objectContaining({
         labels,
         name: expect.anything()
       }),
-      machineSize: defaultAzureMachineType
+      machineSize: defaultAzureMachineType,
+    }, false)
+
+    expect(onSuccess).toHaveBeenCalled()
+  })
+
+  it('sends the proper leo API call in the case of a persistent disk', async () => {
+    // Arrange
+    const createFunc = jest.fn()
+    const runtimeFunc = jest.fn(() => ({
+      create: createFunc,
+      details: jest.fn()
+    }))
+    Ajax.mockImplementation(() => ({
+      ...defaultAjaxImpl,
+      Runtimes: {
+        runtimeV2: runtimeFunc
+      }
     }))
 
+    // Act
+    // wrapping component init-time stateful side-effects with act()
+    await act(async () => {
+      await render(h(AzureComputeModalBase, persistentDiskModalProps))
```

Comment:
render doesn't return a promise, so unless the components `useOnMount` does something with side effects `await` isn't needed here.  Would be good to just check if this component can render without waiting and warnings